src/pages/news/news-add/news-add.js

```diff
@@ -1,273 +1,163 @@
-    setAuthorPhoto,
-    setNewsImage,
-    setNewsVideo,
-    ukSetAuthor,
-    ukSetText,
-    ukSetTitle,
-    enSetAuthor,
-    enSetText,
-    enSetTitle
+    tabsValue,
+    checkboxes,
+    preferredLanguages,
+    setPreferredLanguages,
+    languageCheckboxes,
+    handleTabsChange,
+    createArticle
   } = useNewsHandlers();
 
-  const newsSaveHandler = async (e) => {
-    e.preventDefault();
-    const news = {
-      author: {
-        name: [
-          {
-            lang: languages[0],
-            value: e.target.ukAuthorName.value
-          },
-          {
-            lang: languages[1],
-            value: e.target.enAuthorName.value
-          }
-        ],
-        image: {
-          small: e.target.authorPhoto.value
-        }
-      },
-      title: [
-        {
-          lang: languages[0],
-          value: e.target.ukTitle.value
-        },
-        {
-          lang: languages[1],
-          value: e.target.enTitle.value
-        }
-      ],
-      text: [
-        {
-          lang: languages[0],
-          value: e.target.ukText.value
-        },
-        {
-          lang: languages[1],
-          value: e.target.enText.value
-        }
-      ],
-      images: {
-        primary: {
-          medium: e.target.newsImage.value
-        },
-        additional: {
-          large: 'Test_additional_photo'
-        }
-      },
-      date: new Date().toISOString()
-    };
-    dispatch(addArticle(news));
-  };
+  useEffect(() => {
+    const prefLanguages = [];
+    Object.keys(checkboxes).forEach((key) => {
+      if (checkboxes[key] === true) {
+        prefLanguages.push(key);
+      }
+    });
+    setPreferredLanguages(prefLanguages);
+  }, [checkboxes, setPreferredLanguages]);
 
-  const entertaimentOptions = [
-    {
-      id: 'authorPhoto',
-      className: classes.textfield,
-      variant: 'outlined',
-      label: 'Фото автора',
-      authorPhoto,
-      handler: (e) => setAuthorPhoto(e.target.value),
-      required: true
-    },
-    {
-      id: 'newsImage',
-      className: classes.textfield,
-      variant: 'outlined',
-      label: 'Головне зображення',
-      newsImage,
-      handler: (e) => setNewsImage(e.target.value),
-      required: true
-    },
-    {
-      id: 'newsVideo',
-      className: classes.textfield,
-      variant: 'outlined',
-      label: 'Посилання на відео',
-      newsVideo,
-      handler: (e) => setNewsVideo(e.target.value)
-    }
-  ];
+  const langValues = languages.map((lang) => ({
+    [`${lang}AuthorName`]: '',
+    [`${lang}Title`]: '',
+    [`${lang}Text`]: ''
+  }));
 
-  const ukNewsOptions = [
-    {
-      id: 'ukAuthorName',
-      className: classes.textfield,
-      variant: 'outlined',
-      label: 'Автор (укр.)',
-      ukAuthorName,
-      handler: (e) => ukSetAuthor(e.target.value)
+  const formikValues =
+    langValues !== null ? Object.assign(...langValues) : null;
+  const formik = useFormik({
+    initialValues: {
+      ...formikValues,
```

Comment:
If `formikValues` is equal to `null` it will throw an error, please, fix it

Reply:
Fixed!